bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,313 @@
+                || PJLinkDeviceHandler.this.config.refreshMute || PJLinkDeviceHandler.this.config.refreshInputChannel;
+        if (config.refresh > 0 && atLeastOneChannelToBeRefreshed) {
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refresh, TimeUnit.SECONDS);
+        }
+    }
+
+    private void updateDeviceProperties(PJLinkDevice device) throws IOException, AuthenticationException {
+        updateProperty(PJLinkDeviceBindingConstants.PARAMETER_AUTHENTICATION_REQUIRED,
+                device.getAuthenticationRequired().toString());
+
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_NAME, device.getName());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_NAME, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MANUFACTURER, device.getManufacturer());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MANUFACTURER, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MODEL, device.getModel());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MODEL, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_COMBINED_ID, device.getFullDescription());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_COMBINED_ID, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_CLASS, device.getPJLinkClass());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve class information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_CLASS, e.toString());
+        }
+        try {
+            Map<ErrorStatusDevicePart, ErrorStatusQueryResponseState> errorStatus = device.getErrorStatus();
+            for (Map.Entry<ErrorStatusDevicePart, ErrorStatusQueryResponseState> entry : errorStatus.entrySet()) {
+                String key = entry.getKey().getCamelCaseText();
+                String value = entry.getValue().getText();
+                updateProperty(PJLinkDeviceBindingConstants.PARAMETER_ERROR_STATUS + key, value);
+            }
+        } catch (ResponseException e) {
+            // okay, cannot retrieve class information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_ERROR_STATUS, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_LAMP_HOURS, device.getLampHours());
```

Comment:
This should not be a property but an information to put in a channel.

Reply:
The issue I see with that is that a PJLink projector does not necessarily have only one lamp. 
The string then looks something like 
```
8259 1
13448 0
...
```
which means that lamp one has 8259 hours usage time and is active, lamp two has 13448 hours and is not active, ...
The only way I can imagine to have this as channels it to create channels dynamically based on the number of lamps present in the projector. Sounds like a rather overengineered solution for this (for most users) less crucial bit of functionality.

Reply:
I am not aware of such projectors with several lamps !
I imagine you could define 2 channels, one for lamp 1 and one for lamp 2.
One interest of having a channel is that the user could use this information in a rule.

Reply:
Better idea: just one channel with a channel parameter to set the lamp number. 
Maybe this feature is more to keep trace of duration for the replaced lamps?

Reply:
I assumed the multi-lamp feature is intended for professional high-end projectors which might have redundant/backup lamps (for example in cinemas). In the standard specification it's not even limited to any number, it could even be 20 or more lamps (theoretically :-)). So the idea with the channel parameter sounds appropriate to me :+1: However that might take some time for me to implement (new concept for me)

You're absolutely right, for example users could use this information to know when a lamp comes close to its expected lifetime.

Reply:
Actually there is a limit in the PJlink spec  :-)

> Maximum value of n is 8.

I've experimented with the channel parameter approach and it works in general, but some things like validation of the channel configuration (when and how to do it?) and how to property treat those additional configured channels in handleCommand are not clear to me.

I propose that I add another pull request after the binding is merged, what is your opinion on that?

Reply:
Pull request added for this functionality: https://github.com/openhab/openhab2-addons/pull/6063